### PR TITLE
[RHPAM-4889] - adjustments to make it easier to consume from openshift env vars

### PR DESCRIPTION
**Pull request by porcelli**
**Thank you for submitting this pull request**

**JIRA**: _(please edit the JIRA link if it exists)_ 

[RHPAM-4889](https://issues.redhat.com/browse/RHPAM-4889)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* paste the link(s) from GitHub here
* link 2
* link 3 etc.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by porcelli** [comment]
@mareknovotny can you help me understanding the checks failure?

**Comment by mareknovotny** [comment]
```-------------------------------------------------------------
[2024-02-21T23:42:44.605Z] [ERROR] COMPILATION ERROR : 
[2024-02-21T23:42:44.605Z] [INFO] -------------------------------------------------------------
[2024-02-21T23:42:44.605Z] [ERROR] error reading /home/jenkins/.m2/repository/com/google/http-client/google-http-client/1.23.0/google-http-client-1.23.0.jar; zip file is empty
[2024-02-21T23:42:44.605Z] [ERROR] /home/jenkins/workspace/KIE/main/pullrequest/jbpm-wb-main.pr/bc/kiegroup_jbpm-work-items/google-tasks-workitem/target/generated-sources/java/org/jbpm/process/workitem/google/tasks/GetTasksWorkitemHandler.java:[16,1] cannot access org.jbpm.process.workitem.google.tasks
[2024-02-21T23:42:44.605Z]   zip END header not found
[2024-02-21T23:42:44.605Z] [ERROR] /home/jenkins/workspace/KIE/main/pullrequest/jbpm-wb-main.pr/bc/kiegroup_jbpm-work-items/google-tasks-workitem/target/generated-sources/java/org/jbpm/process/workitem/google/tasks/GoogleTasksAuth.java:[34,8] cannot access java.lang
[2024-02-21T23:42:44.605Z]   zip END header not found
[2024-02-21T23:42:44.605Z] [ERROR] /home/jenkins/workspace/KIE/main/pullrequest/jbpm-wb-main.pr/bc/kiegroup_jbpm-work-items/google-tasks-workitem/target/generated-sources/java/org/jbpm/process/workitem/google/tasks/TaskInfo.java:[23,8] cannot access org.jbpm.process.workitem.google
[2024-02-21T23:42:44.605Z]   zip END header not found
[2024-02-21T23:42:44.605Z] [ERROR] /home/jenkins/workspace/KIE/main/pullrequest/jbpm-wb-main.pr/bc/kiegroup_jbpm-work-items/google-tasks-workitem/target/generated-sources/java/org/jbpm/process/workitem/google/tasks/AddTaskWorkitemHandler.java:[56,8] cannot access org.jbpm.process.workitem
[2024-02-21T23:42:44.605Z]   zip END header not found
[2024-02-21T23:42:44.605Z] [INFO] 5 errors 
[2024-02-21T23:42:44.605Z] [INFO] -------------------------------------------------------------
[2024-02-21T23:42:44.605Z] [INFO] Compiling 3 source files to /home/jenkins/workspace/KIE/main/pullrequest/jbpm-wb-main.pr/bc/kiegroup_jbpm-work-items/google-translate-workitem/target/classes
[2024-02-21T23:42:44.605Z] [INFO] -------------------------------------------------------------
[2024-02-21T23:42:44.605Z] [ERROR] COMPILATION ERROR : 
[2024-02-21T23:42:44.605Z] [INFO] -------------------------------------------------------------
[2024-02-21T23:42:44.605Z] [ERROR] error reading /home/jenkins/.m2/repository/com/google/http-client/google-http-client/1.23.0/google-http-client-1.23.0.jar; zip file is empty
[2024-02-21T23:42:44.605Z] [ERROR] /home/jenkins/workspace/KIE/main/pullrequest/jbpm-wb-main.pr/bc/kiegroup_jbpm-work-items/google-translate-workitem/target/generated-sources/java/org/jbpm/process/workitem/google/translate/TranslateWorkitemHandler.java:[16,1] cannot access org.jbpm.process.workitem.google.translate
[2024-02-21T23:42:44.605Z]   zip END header not found
[2024-02-21T23:42:44.605Z] [ERROR] /home/jenkins/workspace/KIE/main/pullrequest/jbpm-wb-main.pr/bc/kiegroup_jbpm-work-items/google-translate-workitem/target/generated-sources/java/org/jbpm/process/workitem/google/translate/DetectLanguageWorkitemHandler.java:[59,8] cannot access java.lang
[2024-02-21T23:42:44.605Z]   zip END header not found
[2024-02-21T23:42:44.605Z] [ERROR] /home/jenkins/workspace/KIE/main/pullrequest/jbpm-wb-main.pr/bc/kiegroup_jbpm-work-items/google-translate-workitem/target/generated-sources/java/org/jbpm/process/workitem/google/translate/GoogleTranslateAuth.java:[21,8] cannot access org.jbpm.process.workitem.google
[2024-02-21T23:42:44.605Z]   zip END header not found
[2024-02-21T23:42:44.605Z] [INFO] 4 errors 
[2024-02-21T23:42:44.605Z] [INFO] -------------------------------------------------------------
```

these errors come as dependencies are wrongly downloaded imo. Therefore for continued build there are compilation errors.

I can see this error for the first time. We need to try to re-run this to see if it shows again.

jenkins retest this please

**Comment by mareknovotny** [comment]
so this CI PR repository is now green, GHA checks have some problems with installing of dependencies from sources, not related. Trigerring FDB to also see downstream is ok with this change.
jenkins do fdb

**Comment by porcelli** [comment]
@mareknovotny FDB has failed, but doesn't look like  it’s related to the changes

**Comment by mareknovotny** [comment]
ok, now FDB green too. Please add backport labels if this need to be on supported branches too 7.67.x and 7.67.x-blue ;)

**Review by tomasdavidorg — APPROVED** [review]
Looks good to me, just minor comment.

File: jbpm-wb-kie-server/jbpm-wb-kie-server-backend/src/main/java/org/jbpm/workbench/ks/integration/KieServerQueryDefinitionLoader.java [hunk 1]
```diff
@@ -41,54 +41,80 @@ public class KieServerQueryDefinitionLoader {
...
 
+    static final String JBPM_WB_QUERY_MODE = "jbpm.wb.querymode";
+    enum QueryMode {
+        DEFAULT,
+        STRICT;
+
+        static QueryMode convert(final String mode) {
+            try {
+                return QueryMode.valueOf(mode.toUpperCase());
+            } catch (IllegalArgumentException ignored) {
```
[the remark this answers is not in the document]
Comment (mareknovotny) [reply jbpm-wb-kie-server/jbpm-wb-kie-server-backend/src/main/java/org/jbpm/workbench/ks/integration/KieServerQueryDefinitionLoader.java:52]:
```suggestion
            } catch (IllegalArgumentException ignored) {
                LOGGER.warn("Invalid Argument provided. DEFAULT query mode will be used.");
```

**Review by mareknovotny — APPROVED** [review]

**Comment by mareknovotny** [comment]
jenkins retest this please

**Comment by mareknovotny** [comment]
jenkins retest this please
jenkins do fdb

**Review by gmunozfe — APPROVED** [review]
Looks good to me, well done @porcelli !

**Pull request merged**